nncf/quantization/quantization_metric.py

```diff
@@ -0,0 +1,273 @@
+            for column_name in self.header:
+                drow[column_name] = self.stat[column_name][p]
+            drow[self.PARAMS_STR] = p
+            row = [drow[h] for h in self.header]
+            data.append(row)
+        data[0] =[self.PARAMS_STR] + header_for_table
+        table.add_rows(data)
+        return table
+
+class MemoryСostMetric(BaseMetric):
+    """
+
+    This metric considers how many times memory consumption for network weights will decrease.
+    
+    """
+    PARAMS_STR = 'params'
+    NAME_STR = 'MemoryCost'
+
+    EXPECTED_MEMORY_CONSUMPTION_DECREASE_STR = 'memory consumption decrease'
+    SIZE_MEMORY_FP_WEIGHTS_STR = 'size memory for fp weights (bit)'
+    SIZE_MEMORY_COMPRESSED_WEIGHTS_STR = 'size memory for quantized weights (bit)'
+
+    def __init__(self, compressed_model: NNCFNetwork, weights_quantizers):
+        super().__init__()
+        self._compressed_model = compressed_model
+        self._weights_quantizers = weights_quantizers
+        self.header = [self.EXPECTED_MEMORY_CONSUMPTION_DECREASE_STR]
+        self.stat = {}
+
+    def collect(self):
+        self.stat[self.SIZE_MEMORY_FP_WEIGHTS_STR] = 0
+        self.stat[self.SIZE_MEMORY_COMPRESSED_WEIGHTS_STR] = 0
+        fp_num_bits = 32
+        nncf_modules = self._compressed_model.get_nncf_modules()
+
+        for scope_module, nncf_module in nncf_modules.items():
+            count_el = np.prod(nncf_module.weight.shape)
+            self.stat[self.SIZE_MEMORY_FP_WEIGHTS_STR] += count_el * fp_num_bits
+            status, quantizer = self._get_quantizer_for_scope(scope_module)
+            if status > 0:
+                num_bits = quantizer.num_bits
+                self.stat[self.SIZE_MEMORY_COMPRESSED_WEIGHTS_STR] += count_el * num_bits
+            else:
+                self.stat[self.SIZE_MEMORY_COMPRESSED_WEIGHTS_STR] += count_el * fp_num_bits
+        self.stat[self.EXPECTED_MEMORY_CONSUMPTION_DECREASE_STR] = self.stat[self.SIZE_MEMORY_FP_WEIGHTS_STR] /\
+             self.stat[self.SIZE_MEMORY_COMPRESSED_WEIGHTS_STR]
+
+    def _get_quantizer_for_scope(self, scope):
+        for weight_quantizer_id, quantizer in self._weights_quantizers.items():
+            if weight_quantizer_id.get_scope() == scope:
+                return True, quantizer
+        return False, None
+
+    def get_metric_table(self):
+        table = Texttable()
+        data = [self.header]
+        drow = {h: 0 for h in self.header}
+        for column_name in self.header:
+            drow[column_name] = self.stat[column_name]
+        row = [drow[h] for h in self.header]
+        data.append(row)
+        table.add_rows(data)
+        return table
+
+
+class ShareEdgesQuantizedDataPath(BaseMetric):
+    """
+
+    This metric calculates the percentage of quantized edges relative to the total number of edges
+    in the original network graph. Quantized edge is an edge of a compressed graph through which a
+    quantized activation tensor passes.
```

Comment:
A tensor passed through *any* edge. Edges *are* representations of tensors. 
`"Quantized edge" is an edge representing a quantized activation tensor.`